File: cloudus-bugsmash/matrix_transpose - 2/c/matrix.c
#include <stdio.h>

int main() {
    int rows = 2, cols = 3;
    int matrix[2][3] = {{1, 2, 3}, {4, 5, 6}};
    int transpose[2][3]; 
    
    for(int i = 0; i <= rows; ++i) 
        for(int j = 0; j <= cols; ++j) 
            transpose[i][j] = matrix[j][i]; 

    
    printf("Transpose of the matrix:\n");
    for(int i = 0; i < rows; ++i) { 
        for(int j = 0; j < cols; ++j) 
            printf("%d ", transpose[i][j]);
        printf("\n");
    }
    return 0;
}

File: cloudus-bugsmash/matrix_transpose - 2/java/matrix.java
public class MatrixTranspose {
    public static void main(String[] args) {
        int[][] matrix = {
            {1, 2, 3},
            {4, 5}  
        };

        int rows = matrix.length;
        int cols = matrix[0].length;
        int[][] transpose = new int[rows][cols]; 

       
        for (int i = 0; i <= rows; i++) { 
            for (int j = 0; j <= cols; j++) { 
                transpose[j][i] = matrix[i][j]; 
            }
        }

       
        System.out.println("Transpose of the matrix:");
        for (int i = 0; i < rows; i++) { 
            for (int j = 0; j < cols; j++) { 
                System.out.print(transpose[i][j]); 
            }
            System.out.println();
        }
    }
}
